jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -100,9 +106,16 @@ public interface HttpChannel extends Invocable
     /**
      * Recycle the HttpChannel, so that a new cycle of calling {@link #setHttpStream(HttpStream)},
      * {@link #onRequest(MetaData.Request)} etc. may be performed on the channel.
+     * @see #init()
      */
     void recycle();
 
+    /**
+     * Initialize the HttpChannel when a new cycle of request handling begins.
+     * @see #recycle()
+     */
+    void init();
```

Comment:
See comments by @gregw at https://github.com/jetty/jetty.project/pull/11301#discussion_r1462556022 for concerns about naming here.